pkg/cli/startbuild/startbuild.go

```diff
@@ -513,6 +526,120 @@ func (o *StartBuildOptions) streamBuildLogs(build *buildv1.Build) error {
 	return err
 }
 
+// checkNonExistantResources checks for Resources (config maps, secrets, etc) listed in the BuildConfig
+// and returns error in case any of them are missing, or in case of unexpected behavior
+func (o *StartBuildOptions) checkNonExistantResources() error {
+	buildConfig, err := o.BuildClient.BuildConfigs(o.Namespace).Get(context.TODO(), o.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	cms := make([]string, 0, 3)
+	secrets := make([]string, 0, 3)
```

Comment:
Is there a compelling reason to set an initial capacity here? I don't think we lose much with an initial capacity of 0:
```suggestion
	cms := []string{}
	secrets := []string{}
```